Cython/Compiler/ExprNodes.py

```diff
@@ -3246,7 +3282,33 @@ def infer_type(self, env, iterator_type=None):
             return fake_index_node.infer_type(env)
 
     def analyse_types(self, env):
+        def remove_const(item_type):
```

Comment:
Still think this would be better as a non-nested function given that it doesn't capture anything